src/accelerate/utils/deepspeed.py

```diff
@@ -12,58 +12,26 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from ..optimizer import AcceleratedOptimizer
-from .imports import is_apex_available, is_deepspeed_available
-
-
-if is_deepspeed_available():
-    from deepspeed import DeepSpeedEngine
-
-if is_apex_available():
-    from apex import amp
-
-
-class DeepSpeedEngineWrapper(DeepSpeedEngine):
-    """
-    Wrapper over deepspeed.DeepSpeedEngine object
-    """
-
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
+from accelerate.scheduler import AcceleratedScheduler
 
-        # overwriting micro_steps for user's gradient_accumulation
-        self.micro_steps = -1
+from ..optimizer import AcceleratedOptimizer
 
-    def step(self, lr_kwargs=None):
-        """DeepSpeedEngine.step() without `micro_steps` update & no profiling"""
-        if self.is_gradient_accumulation_boundary():  # it shouldn't matter whether we keep this line or not
-            if self.progressive_layer_drop:
-                self.progressive_layer_drop.update_state(self.global_steps)
 
-            self._take_model_step(lr_kwargs)
+class DeepSpeedEngineWrapper:
+    def __init__(self, engine):
+        self.engine = engine
 
     def backward(self, loss):
-        """DeepSpeedEngine.backward() with with no loss scaling; no profiling but with `micro_steps` update"""
-
-        if self.zero_optimization():
-            self.optimizer.is_gradient_accumulation_boundary = self.is_gradient_accumulation_boundary()
-            self.optimizer.backward(loss)
-        elif self.amp_enabled():
-            # AMP requires delaying unscale when inside gradient accumulation boundaries
-            # https://nvidia.github.io/apex/advanced.html#gradient-accumulation-across-iterations
-            delay_unscale = not self.is_gradient_accumulation_boundary()
-            with amp.scale_loss(loss, self.optimizer, delay_unscale=delay_unscale) as scaled_loss:
-                scaled_loss.backward()
-        elif self.fp16_enabled():
-            self.optimizer.backward(loss)
-        else:
-            loss.backward()
-
-        if self.enable_backward_allreduce:
-            self.allreduce_gradients()
+        # runs backpropagation and handles mixed precision
+        self.engine.backward(loss)
 
-        # this will ensure deepspeed gradient_accumulation matches user's accumulation
-        self.micro_steps += 1
+        # gradient accumulation check
+        # gradient clipping
+        # optimizer step
+        # zero grad
+        # checking overflow
+        # lr_scheduler step
+        self.engine.step()
```

Comment:
Not sure I understand the comments. Is it everything that happens in this step? If that's the case, can you add a comment stating it at the top?